go/database/mpt/proof.go

```diff
@@ -297,3 +298,121 @@ func createPathFromKeyPrefix(key common.Key, nibbles uint8) Path {
 	copy(res.path[:], key[:])
 	return res
 }
+
+// NewWitnessProof creates a new witness proof.
+func NewWitnessProof() witness.Proof {
+	return witnessProofAdapter{WitnessProof{proofDb{}}}
+}
+
+// WitnessProofToNodes converts the witness proof to a list of strings.
+// Each string is an RLP node of the witness proof.
+func WitnessProofToNodes(proof witness.Proof) []string {
```

Comment:
I think this should be a member function of the proof instead of a stand-alone function.
It might also be part of the proof interface.

Reply:
it would loos genericity, the idea was that `witness.Proof` serves for all schemes while the nodes are only in MPT.

I was also thinking about something more general, like a method `Serialise/Deserialise`, which would work with `[]byte` etc, it could then go to the interface.